http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -234,15 +239,28 @@ class LeoPubsubMessageSubscriber[F[_]: Async: Timer: ContextShift: Concurrent](
     for {
       runtimeOpt <- clusterQuery.getClusterById(msg.runtimeId).transaction
       runtime <- runtimeOpt.fold(
-        Async[F].raiseError[Runtime](PubsubHandleMessageError.ClusterNotFound(msg.runtimeId, msg))
-      )(Async[F].pure)
+        F.raiseError[Runtime](PubsubHandleMessageError.ClusterNotFound(msg.runtimeId, msg))
+      )(F.pure)
       _ <- if (runtime.status != RuntimeStatus.Deleting)
-        Async[F].raiseError[Unit](
+        F.raiseError[Unit](
           PubsubHandleMessageError.ClusterInvalidState(msg.runtimeId, runtime.projectNameString, runtime, msg)
         )
-      else Async[F].unit
+      else F.unit
       runtimeConfig <- RuntimeConfigQueries.getRuntimeConfig(runtime.runtimeConfigId).transaction
-      _ <- runtimeConfig.cloudService.interpreter.deleteRuntime(DeleteRuntimeParams(runtime))
+      _ <- if (runtimeConfig.cloudService == CloudService.GCE) {
+        googleComputeService
+          .deleteInstance(runtime.googleProject, config.gceZoneName, InstanceName(runtime.runtimeName.asString))
```

Comment:
I think you should call `CloudService.GCE.interpreter.deleteRuntime()` instead of `googleComputeService.deleteInstance()` -- the GCE interpreter might orchestrate other Google calls as well.

Reply:
Reason why I'm not using `GceInterpreter` is we need an `Operation` back from the `deleteRuntime`, which doesn't apply to current `DataprocInterpreter`